### PR TITLE
Initial support for detecting modified libraries

### PR DESCRIPTION
**Pull request by johnmccutchan**
- Roll Dart to 59e67cc36d0877949719160de8959a11db4a8027
- Pass in resolved uri to Dart_LoadScript, Dart_LoadLibrary, and Dart_LoadSource API calls.
- Register a FileModifiedCallback with the VM.

### CODE REVIEW

**Comment by johnmccutchan** [comment]
@Hixie @abarth

File: flutter/tonic/dart_library_loader.h [hunk 1]
```diff
@@ -31,11 +31,14 @@ class DartLibraryLoader {
...
 
   // TODO(dart): This can be called both on the main thread from application
-  // solates or from the handle watcher isolate thread.
+  // isolates or from the handle watcher isolate thread.
   static Dart_Handle HandleLibraryTag(Dart_LibraryTag tag,
                                       Dart_Handle library,
                                       Dart_Handle url);
 
+  static bool IsSourceUrlModified(const char* source_url,
+                                  int64_t since);
```
Comment (abarth) [inline flutter/tonic/dart_library_loader.h]: Is there any reason to put this function here?  It seems like a general function that doesn't have anything to do with Dart or Tonic.

**Comment by abarth** [comment]
LGTM except for the misplaced function.

File: flutter/tonic/dart_library_loader.h (hunk 1 above)
Comment (johnmccutchan) [inline flutter/tonic/dart_library_loader.h]: Moved to a private namespace in dart_init.cc

**Comment by abarth** [comment]
Thanks

**Pull request merged**